overlord/devicestate/handlers.go

```diff
@@ -91,6 +92,27 @@ func (m *DeviceManager) doSetModel(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// unmark no-longer required snaps
+	requiredSnaps := getAllRequiredSnapsForModel(new)
+	snapStates, err := snapstate.All(st)
+	for snapName, snapst := range snapStates {
+		// TODO: remove this type restriction once we remodel
+		//       bases/kernels/gadgets and add tests that ensure
+		//       that the required flag is properly set/unset
+		typ, err := snapst.Type()
+		if err != nil {
+			return err
+		}
+		if typ != snap.TypeApp && typ != snap.TypeBase {
+			continue
+		}
```

Comment:
I was staring at canRemove,   it seems in principle will need to do something also for allowing removing core if it's not the bootable base and nothing is using it.  That would mean that we would allow removing the flag here as well.
It can just be a TODO for now.

I agree though kernel, gadget would need further work both here and in canRemove.

Reply:
thanks for the new comment